### ISSUE TITLE
Can't open specific file that contains pivot tables

### ISSUE THREAD

**Issue by viogroza** [ClosedXML/ClosedXML#2498]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [x] Yes
- [ ] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**

0.104.1, latest develop (as of 13 Nov 2024)
the problem seems to be introduced in 0.102.1

**What is the current behavior?**

Unable to open the attached xlsx file

**What is the expected behavior or new feature?**

Attached xlsx file should open properly

**Is this a regression from the previous version?**

It's a regression introduced in 0.102.1.0
In 0.100.3 opening the file worked

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

Open the attached file. The stack trace is 
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at ClosedXML.Excel.IO.PivotTableCacheDefinitionPartReader.ParsePivotSourceReference(PivotTableCacheDefinitionPart pivotTableCacheDefinitionPart)
   at ClosedXML.Excel.IO.PivotTableDefinitionPartReader.<>c__DisplayClass1_0.<Load>b__1(XLPivotCache ps)
   at System.Linq.Enumerable.TryGetFirst[TSource](IEnumerable`1 source, Func`2 predicate, Boolean& found)
   at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
   at ClosedXML.Excel.IO.PivotTableDefinitionPartReader.Load(WorkbookPart workbookPart, Dictionary`2 differentialFormats, PivotTablePart pivotTablePart, WorksheetPart worksheetPart, XLWorksheet ws, LoadContext context)
   at ClosedXML.Excel.XLWorkbook.LoadSpreadsheetDocument(SpreadsheetDocument dSpreadsheet)
   at ClosedXML.Excel.XLWorkbook.LoadSheets(String fileName)
   at ClosedXML.Excel.XLWorkbook.Load(String file)
   at ClosedXML.Excel.XLWorkbook..ctor(String file, LoadOptions loadOptions)
   at ClosedXML.Excel.XLWorkbook..ctor(String file)
   at Program.<Main>$(String[] args) in C:
   
   
![image](https://github.com/user-attachments/assets/b39f0b44-53f2-439b-ad16-276b6001eeea)

I noticed that there are a few similar reported issues, but they don't have any excel file attached
https://github.com/ClosedXML/ClosedXML/issues/2309
https://github.com/ClosedXML/ClosedXML/issues/2063

This one is is also similar, has attachment but the problem is solved on latest develop
https://github.com/ClosedXML/ClosedXML/issues/2272

The crash reported by this ticket is still happening on latest develop (as of 13 Nov 2024)

**Code to reproduce problem:**
```c#
public void Main()
{
  //See attached sample below, it should reproduce the problem
   using var a = new XLWorkbook(fileName);
}
```

- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
[FilterPivotTableXTests.xlsx](https://github.com/user-attachments/files/17734584/FilterPivotTableXTests.xlsx)
[ClosedXMLFailsToOpen.zip](https://github.com/user-attachments/files/17734589/ClosedXMLFailsToOpen.zip)

**Comment by mcastillob** [comment]
Hi, 

The problem is with the external cache. There is no support apparently. The error occurs in these cases.

![image](https://github.com/user-attachments/assets/e32b758a-9fa5-47cf-adcd-5054db4cffe4)

**Comment by jahav** [comment]
Fixed in #2585.

Note that the workbook is using power pivot/data model. The only thing I want from such workbooks is not to crash on load. Saving/manipulation... that's not even on the long term roadmap.
